### PR TITLE
Fix incorrect compilation flag in doc example.

### PR DESCRIPTION
**Pull request by frewsxcv**
I agree to license my contributions to each file under the terms
given at the top of each file I changed.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -136,7 +136,7 @@ mod rsa;
 
 pub mod signature;
 
-#[cfg(any(feature = "use_heap", test))]
+#[cfg(feature = "use_heap")]
```
Comment (briansmith) [inline src/lib.rs]: Consider the case where the library will be linked into an executable that doesn't have `malloc`. That is what disabling `use_heap` is supposed to allow.

But, we still want people to be able to run the *ring* test suite for that configuration, in another executable (the one generated by `cargo test`. Right now, that means they have to have a working `malloc`. In the future, we'll hopefully remove this requirement.

tl;dr: I think the original condition is what we want.

File: src/digest/digest.rs [hunk 2]
```diff
@@ -221,7 +221,7 @@ impl Clone for Context {
 /// # Examples:
 ///
 /// ```
-/// # #[cfg(any(feature = "use_heap", test))]
+/// # #[cfg(feature = "use_heap")]
```
Comment (briansmith) [inline src/digest/digest.rs]: I agree this and the following identical change are good.

**Comment by briansmith** [comment]
Thanks!
